typescript/cli/src/deploy/warp.ts

```diff
@@ -298,9 +302,19 @@ async function fetchBaseTokenMetadata(
       { token: address },
     );
     return adapter.getMetadata();
+  } else if (base.type === TokenType.collateralVault && vaultAddress) {
+    // If it's a collateralVault type, query the vault's metadata.
+    log(`Fetching vault metadata for ${address} on ${chainName}}`);
+    // ERC4626 is inherits from ERC20, so it's okay to use EvmTokenAdapter
+    const adapter = new EvmTokenAdapter(
+      chainName,
+      MultiProtocolProvider.fromMultiProvider(multiProvider),
+      { token: vaultAddress },
+    );
+    return adapter.getMetadata();
```

Comment:
if you reuse the token config, we dont need this iiuc

Reply:
Removed